drivers/docker/driver.go

```diff
@@ -618,12 +618,15 @@ func (d *Driver) containerBinds(task *drivers.TaskConfig, driverConfig *TaskConf
 		// Paths inside task dir are always allowed, Relative paths are always allowed as they mount within a container
 		// When a VolumeDriver is set, we assume we receive a binding in the format volume-name:container-dest
 		// Otherwise, we assume we receive a relative path binding in the format relative/to/task:/also/in/container
-		if localBindVolume {
+		if taskLocalBindVolume {
 			parts[0] = expandPath(task.TaskDir().Dir, parts[0])
+		} else {
+			// Resolve dotted path segments
+			parts[0] = filepath.Clean(parts[0])
```

Comment:
Let's add a test to exercise this path given it's part of 0.8.7.  That being said, if `part[0]` is to be treated as a volume name and not a path, I think we ought to pass the string verbatim without modification.